NetCord/Gateway/GatewayClient.cs

```diff
@@ -17,71 +20,385 @@ public partial class GatewayClient : WebSocketClient, IEntity
     private readonly IGatewayCompression _compression;
     private readonly bool _disposeRest;
 
+    /// <summary>
+    /// The ready event is dispatched when a client has completed the initial handshake with the gateway (for new sessions).
+    /// The ready event can be the largest and most complex event the gateway will send, as it contains all the state required for a client to begin interacting with the rest of the platform.
+    /// </summary>
     public event Func<ReadyEventArgs, ValueTask>? Ready;
+
+    /// <summary>
+    /// Sent when an application command's permissions are updated.
+    /// The inner payload is an <see cref="ApplicationCommandPermission"/> object.
+    /// </summary>
     public event Func<ApplicationCommandPermission, ValueTask>? ApplicationCommandPermissionsUpdate;
+
+    /// <summary>
+    /// Sent when a rule is created.
+    /// The inner payload is an <see cref="AutoModerationRule"/> object.
+    /// </summary>
     public event Func<AutoModerationRule, ValueTask>? AutoModerationRuleCreate;
+
+    /// <summary>
+    /// Sent when a rule is updated.
+    /// The inner payload is an <see cref="AutoModerationRule"/> object.
+    /// </summary>
     public event Func<AutoModerationRule, ValueTask>? AutoModerationRuleUpdate;
+
+    /// <summary>
+    /// Sent when a rule is deleted.
+    /// The inner payload is an <see cref="AutoModerationRule"/> object.
+    /// </summary>
     public event Func<AutoModerationRule, ValueTask>? AutoModerationRuleDelete;
+
+    /// <summary>
+    /// Sent when a rule is triggered and an action is executed (e.g. when a message is blocked).
+    /// </summary>
     public event Func<AutoModerationActionExecutionEventArgs, ValueTask>? AutoModerationActionExecution;
+
+    /// <summary>
+    /// Sent when a new guild channel is created, relevant to the bot.
```

Comment:
Discord's documentation just says 'relevant to the current user' (or something to that tone), so I actually don't know what it could mean past something in its scope.